File: PythonNBA/scripts/self_tune.py
# scripts/self_tune.py  --  thin wrapper around core/self_tune.py
import sys, os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..'))

import defaults
import core.self_tune as _tune

_tune._configure(defaults)

_tune.HCA_CLAMP_LO = defaults.HCA_CLAMP_LO
_tune.HCA_CLAMP_HI = defaults.HCA_CLAMP_HI
_tune.HCA_VAR_FLOOR = defaults.HCA_VAR_FLOOR

# Re-export public API
from core.self_tune import tune_weights, compute_residual_var

File: NCAA/scripts/self_tune.mjs
import { createSelfTune } from "../../core-js/self_tune.mjs";
import { DEFAULT_W, DEFAULT_W_VAR, BAYES_HYPER } from "./defaults.mjs";

const { tuneWeights, computeResidualVar } = createSelfTune({
  DEFAULT_W,
  DEFAULT_W_VAR,
  BAYES_HYPER,
  hcaClamp: [2.0, 6.0],
  hcaVarFloor: 0.3,
});

export { tuneWeights, computeResidualVar };
